BetterPasswordPlugin.inc.php

```diff
@@ -208,7 +198,94 @@ function checkPassword($hookName, $args) {
 			}
 		}
 	}
-
+	/**
+	 * Check for newly added blacklists. If ran the first time creates the list of blacklists
+	 * @return boolean false if update failed, true if updated the setting
+	 */	
+	function generateBlacklist() {
+		$prevBlacklist = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+		$updateSettings = false;
+		$newBlacklist = array();
+		foreach ($this->getBlacklists() as $filename) {
+			$newBlacklist[$filename] = sha1_file($filename);
+		}
+		if (is_null($prevBlacklist)) {
+			$updateTempFile = $this->handleTempFile();
+			if ($updateTempFile) {
+				$updateSettings = $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $newBlacklist);
+			}
+		} else {
+			if ($prevBlacklist != $newBlacklist) {
+				$updateTempFile = $this->handleTempFile(false);
+				if($updateTempFile) {
+					$updateSettings = $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $newBlacklist);
+				}
+			} else {
+				$updateSettings = true;
+			}
+		}
+		return (boolean) $updateSettings;
+	}
+        
+	/**
+	 * Creates a temporary file to aggregate all the passwords from the blacklist files if the temporary file doesn't exist
+	 * @return string|null The file path for the temporary file
+	 */
+	function getTempFile() {
+		import('lib.pkp.classes.file.PrivateFileManager');
+		$fileMgr = new PrivateFileManager();
+		$tempFileDir = realpath($fileMgr->getBasePath()) . DIRECTORY_SEPARATOR . 'betterPassword';
+		if (!$fileMgr->fileExists($tempFileDir, 'dir')) {
+			$success = $fileMgr->mkdirtree($tempFileDir);
+			if (!$success) {
+				error_log('ERROR: Unable to create directory' . $tempFileDir);// Files directory wrong configuration?
+				return null;
+			}
+		}
+		if(!file_exists($tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile')) {
+			touch($tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile');
+		}
+		return $tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile';
+	}
+        
+	/**
+	 * Handles the temporary file with the data of password blacklists.
+	 * @param boolean $trustExistingFile Default true, only false when updating temp file
+	 * @return boolean True if operations done to the temp file succeed  
+	 */
+	function handleTempFile($trustExistingFile = true) {
+		$siteDao = DAORegistry::getDAO('SiteDAO');
+		$site = $siteDao->getSite();
+		$minLengthPass = $site->getMinPasswordLength();
+		$tempFilePath = $this->getTempFile();
+		if($trustExistingFile) {
```

Comment:
`$trustExistingFile` doesn't make sense.  If there was no existing blacklist setting in `generateBlacklist()`, the existing file should be nonexistent, and `$trustExistingFile` is true.  You then open the file for append, rather than for write.  But what if the blacklist setting didn't exist in the database, but a file (spurious or malicious) did exist on the system?  You would then append to the existing file.

When the blacklist settings are changed (or the site minimum length setting is changed), you need to truncate and re-create the the blacklist file.  (Opening the file in mode "w".)

The semantics of this method is really to populate or re-populate the tempfile.  There is never an append operation where you trust the existing content.